### PR TITLE
feat(html): minify the document an <iframe srcdoc> holds

### PR DESCRIPTION
**Pull request by aryanraj45**
**Summary**

An `<iframe srcdoc>` holds a whole HTML document, and minifying the page around it left that document untouched — comments, redundant whitespace and quotes all shipped as written. This adds `optimization.minimize.html.minifySrcdoc` (off by default) to minify it with the same options the outer document gets, which is also what an inline `<style>`'s CSS already does.

Like a conditional comment's body it cannot be minified where it is found — the visitor runs inside the document's own parse and the walk state is module-scoped — so the values are recorded, minified, and put back into the source, which is then minified as usual. Unlike a conditional comment they go back into the source rather than the output, because the printer respells an attribute value, so the recorded text is not what the output holds. A nested iframe recurses, and only a document naming the attribute at all pays the extra parse.

**What kind of change does this PR introduce?**

feat

**Did you add tests for your changes?**

Yes — `test/configCases/html/minimize-srcdoc` covers a basic body, both quote delimiters, a nested iframe, an inline `<style>`, a whole document with a doctype, entity handling, an uppercase attribute name, an empty value, an iframe without the attribute, and the word appearing in text.

**Does this PR introduce a breaking change?**

No. The option is off by default, so output is unchanged unless it is turned on.

**If relevant, what needs to be documented once your changes are merged or what have you already documented?**

`optimization.minimize.html.minifySrcdoc` needs an entry alongside the other `optimization.minimize.html` options; the schema carries its description.

**Use of AI**

AI was used to help implement and review this change. Every decision was checked against the code and the output: the emitted page is snapshotted, and the minified `srcdoc` was rendered in Chrome and compared against the original by walking the iframe's inner document, so the nested minification is held to producing the same document

## Summary by CodeRabbit

* **New Features**
  * Added an option to minify HTML embedded in `<iframe srcdoc>` attributes, including nested documents and conditional comments.
  * Supports styles, doctypes, entities, quoting, and case-insensitive attributes while preserving unrelated content.

* **Bug Fixes**
  * Improved conditional-comment handling to preserve conditions and content accurately during minification.

* **Documentation**
  * Added the new HTML optimization option to configuration documentation.

* **Tests**
  * Added coverage for nested markup, varied HTML content, and edge cases.

### CODE REVIEW

File: schemas/WebpackOptions.json [hunk 1]
```diff
@@ -3781,6 +3781,11 @@
           "type": "boolean",
           "added": "5.110.0"
         },
+        "minifySrcdoc": {
+          "description": "Minify the document held in an `<iframe srcdoc>` attribute. Off by default: the body is a whole document of its own (its base URL is `about:srcdoc`), so minifying it is safe, but the attribute is readable from script and a consumer comparing `iframe.srcdoc` byte for byte would see it change.",
+          "type": "boolean",
+          "added": "5.110.0"
+        },
```
[the remark this answers is not in the document]
Comment (aryanraj45) [reply schemas/WebpackOptions.json:3800]: The generated output is already in this PR -281c581e8 includesdeclarations/WebpackOptions.d.ts, schemas/WebpackOptions.check.js and types.d.ts alongside the schema change, and re-running yarn fix:special
produces no further diff

**Comment by alexander-akait** [comment]
Let’s rebase

**Review by alexander-akait — APPROVED** [review]
